src/ServiceManagerBase.sol

```diff
@@ -135,6 +137,114 @@ abstract contract ServiceManagerBase is ServiceManagerBaseStorage {
         _setRewardsInitiator(newRewardsInitiator);
     }
 
+    function migrateOperatorSetIds(uint32[] memory operatorSetsToCreate) external onlyOwner{
+        _migrateCreateOperatorSetIds(operatorSetsToCreate);
+    }
+
+    function migrateToOperatorSets(uint32[][] memory operatorSetIds, address[] memory operators) external onlyOwner {
+        require(!migrationFinalized, "SerivceManager: Migration Already Finalized");
+        _migrateToOperatorSets(operatorSetIds, operators);
+    }
+
+    function finalizeMigration() external onlyOwner{
+        require(!migrationFinalized, "SerivceManager: Migration Already Finalized");
+        migrationFinalized = true;
+    }
+
+    function _migrateCreateOperatorSetIds(uint32[] memory operatorSetIdsToCreate) internal {
+        _avsDirectory.becomeOperatorSetAVS();
+        AVSDirectory(address(_avsDirectory)).createOperatorSets(operatorSetIdsToCreate);
+    }
+
+    function _migrateToOperatorSets(uint32[][] memory operatorSetIds, address[] memory operators) internal {
+        AVSDirectory(address(_avsDirectory)).migrateOperatorsToOperatorSets(operators, operatorSetIds);
+    }
+
+    function getOperatorsToMigrate() public view returns (uint32[] memory operatorSetIdsToCreate, uint32[][] memory operatorSetIds, address[] memory allOperators) {
```

Comment:
Can we add fuzz tests for this function?

Reply:
yeah that's reasonable

Reply:
Added a fuzz test for registering random sets of operators and viewing the migration data. checked the invariants of the _registerRandomOperators [6f37c47](https://github.com/Layr-Labs/eigenlayer-middleware/pull/286/commits/6f37c4710a9e441a386f0334db4924199611f696)